### PR TITLE
ScreenGridLayer: Split into Composite and SubLayer

### LINKED ISSUE #3526: ScreenGridLayer: Fix AttributeManager usage
`ScreenGridLayer`, currently uses `GPUGridAggregator` to create attribute buffers and uses `attributeManger.update()` directly, instead it should pass these buffers to a sub-layer, which should consume these and use AttributeManager to setup attributes. Similar to `GPUGridLayer`.

CC: @Pessimistress

### PR DESCRIPTION
**Pull request by 1chandu**
Fixes #3526 

#### Background
- Split ScreenGridLayer into, 
  1. ScreenGirdLayer (Composite) that handles aggregation
  2. ScreenGridCellLayer (Layer) that handles rendering grid cells.

#### Change List
- ScreenGridLayer: Split into Composite and SubLayer

### CODE REVIEW

File: modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      return false;
+    }
+    // None specified, use default minColor and maxColor
+    return true;
+  }
+
+  _updateUniforms({oldProps, props, changeFlags}) {
+    const newState = {};
+    if (COLOR_PROPS.some(key => oldProps[key] !== props[key])) {
+      newState.shouldUseMinMax = this._shouldUseMinMax();
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js]: Just call `model.setUniforms`

File: modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    // minColor and maxColor not supplied, check if colorRange or colorDomain supplied.
+    // NOTE: colorDomain and colorRange are experimental features, use them only when supplied.
+    if (colorDomain || colorRange) {
+      return false;
+    }
+    // None specified, use default minColor and maxColor
+    return true;
+  }
+
+  _updateUniforms({oldProps, props, changeFlags}) {
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js]: - Avoid using named parameters in internal methods
- `changeFlags` not used

File: modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    if (oldProps.cellSize !== props.cellSize || changeFlags.viewportChanged) {
+      const attributeManager = this.getAttributeManager();
+      attributeManager.invalidateAll();
+    }
+
+    this._updateUniforms({oldProps, props, changeFlags});
+  }
+
+  draw({uniforms}) {
+    const {parameters = {}, maxTexture} = this.props;
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js]: Unnecessary fallback - `Object.assign` handles `undefined`

File: modules/aggregation-layers/src/screen-grid-layer/screen-grid-layer.js [hunk 4]
```diff
@@ -118,6 +88,33 @@ export default class ScreenGridLayer extends Layer {
...
+    const {updateTriggers} = this.props;
+
+    return new ScreenGridCellLayer(
+      this.props,
+      this.getSubLayerProps({
+        id: 'cell-layer',
+        updateTriggers
+      }),
+      {
+        data: {
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/screen-grid-layer/screen-grid-layer.js]: This is going to invalidate all attributes in the sub layer every time viewport changes. Cache the `data` object or supply `instanceCounts` as a top-level prop.

File: modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+const defaultProps = {
+  cellSizePixels: {value: 100, min: 1},
+  cellMarginPixels: {value: 2, min: 0, max: 5},
+
+  colorDomain: null,
+  colorRange: defaultColorRange
+};
+
+export default class ScreenGridCellLayer extends Layer {
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js]: Add standard tests for this layer.

File: modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+    });
+    this.setState({
+      model: this._getModel(gl)
+    });
+  }
+
+  updateState({oldProps, props, changeFlags}) {
+    super.updateState({oldProps, props, changeFlags});
+
+    if (oldProps.cellSize !== props.cellSize || changeFlags.viewportChanged) {
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js]: `cellSize` not declared in `defaultProps`

File: modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js (hunk 2 above)
Comment (1chandu) [reply modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js]: Removed named params, `changeFlags` is used in this function.

File: modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js (hunk 5 above)
Comment (1chandu) [reply modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js]: Added tests, couldn't use auto generated tests, as they contain `data` as array or Null.

File: modules/aggregation-layers/src/screen-grid-layer/screen-grid-layer.js (hunk 4 above)
Comment (1chandu) [reply modules/aggregation-layers/src/screen-grid-layer/screen-grid-layer.js]: updated to cache data prop.

**Pull request merged**